health-services/project/src/main/java/org/egov/project/repository/LocationCaptureRepository.java

```diff
@@ -0,0 +1,133 @@
+        super(producer, namedParameterJdbcTemplate, redisTemplate, selectQueryBuilder, locationCaptureRowMapper, Optional.of("user_location"));
+    }
+
+    /**
+     * Finds user locations based on search criteria and URL parameters.
+     *
+     * @param searchObject The search criteria for user locations.
+     * @param urlParams    The URL parameters including pagination and filtering information.
+     * @return A SearchResponse containing the list of user locations and the total count.
+     */
+    public SearchResponse<UserAction> find(UserActionSearch searchObject, URLParams urlParams) {
+        log.info("Executing find with searchObject: {} and urlParams: {}", searchObject, urlParams);
+
+        String query = "SELECT id, clientreferenceid, tenantid, projectid, latitude, longitude, locationaccuracy, boundarycode, action, createdby, createdtime, lastmodifiedby, lastmodifiedtime, clientcreatedtime, clientlastmodifiedtime, clientcreatedby, clientlastmodifiedby, additionaldetails FROM user_location ul ";
+
+        Map<String, Object> paramsMap = new HashMap<>();
+        List<String> whereFields = GenericQueryBuilder.getFieldsWithCondition(searchObject, QueryFieldChecker.isNotNull, paramsMap);
+        query = GenericQueryBuilder.generateQuery(query, whereFields).toString();
+        query = query.replace("id IN (:id)", "ul.id IN (:id)");
+        query = query.replace("clientReferenceId IN (:clientReferenceId)", "ul.clientReferenceId IN (:clientReferenceId)");
+
+        if (CollectionUtils.isEmpty(whereFields)) {
+            query = query + " WHERE ul.tenantId=:tenantId ";
+        } else {
+            query = query + " AND ul.tenantId=:tenantId ";
+        }
+
+        if (urlParams.getLastChangedSince() != null) {
+            query = query + " AND ul.lastModifiedTime>=:lastModifiedTime ";
+        }
+        paramsMap.put("tenantId", urlParams.getTenantId());
+        paramsMap.put("lastModifiedTime", urlParams.getLastChangedSince());
+
+        try {
+            log.debug("Executing query to fetch total count");
+            Long totalCount = CommonUtils.constructTotalCountCTEAndReturnResult(query, paramsMap, this.namedParameterJdbcTemplate);
+
+            query = query + " ORDER BY ul.id ASC LIMIT :limit OFFSET :offset";
+            paramsMap.put("limit", urlParams.getLimit());
+            paramsMap.put("offset", urlParams.getOffset());
+
+            log.debug("Executing query to fetch user locations: {}", query);
+            List<UserAction> locationCaptureList = this.namedParameterJdbcTemplate.query(query, paramsMap, this.rowMapper);
+
+            log.info("Successfully fetched user locations: {}", locationCaptureList.size());
+            return SearchResponse.<UserAction>builder().response(locationCaptureList).totalCount(totalCount).build();
+        } catch (Exception e) {
+            log.error("Failed to execute query for finding user locations", e);
```

Comment:
error log should be printed, verify

Reply:
verified on local, it will print whole stack trace